### PR TITLE
feat: api videos playlist

### PR DESCRIPTION
**Pull request by mikeallisonJS**
# Description

This adds playlist support for videos, ingests series information from arclight, and adds filters to the video queries for the video api

https://3.basecamp.com/3105655/buckets/26736589/todos/4795311585

# How should this PR be QA Tested?

- [ ] Existing video calls in journeys-admin are unaffected
- [ ] Playlists and filters are available in watch

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged into main

### CODE REVIEW

**Review by gdub01 — APPROVED** [review]
awesome

File: apps/api-videos/db/seed.ts [hunk 1]
```diff
@@ -191,8 +196,16 @@ async function digestContent(
...
 async function digestMediaComponentLanguage(
-  mediaComponentLanguage: MediaComponentLanguage
+  mediaComponentLanguage: MediaComponentLanguage,
+  mediaComponent: MediaComponent
 ): Promise<VideoVariant> {
+  if (mediaComponent.subType === 'series') {
+    return {
+      id: mediaComponentLanguage.refId,
+      languageId: mediaComponentLanguage.languageId.toString(),
+      duration: Math.round(mediaComponentLanguage.lengthInMilliseconds * 0.001)
```
Comment (tataihono) [inline apps/api-videos/db/seed.ts:206]: add `subtitle: []`, and `downloads: []`

Comment (tataihono) [reply apps/api-videos/db/seed.ts:206]: further down and here you should also add `episodeIds: []`

File: apps/api-videos/db/seed.ts [hunk 2]
```diff
@@ -67,21 +68,23 @@ interface Download {
...
+  primaryLanguageId: string
   title: Translation[]
   snippet: Translation[]
   description: Translation[]
   studyQuestions: Translation[][]
   image: string
-  variants: VideoVariant[]
+  variants?: VideoVariant[]
   tagIds: string[]
+  episodeIds?: string[]
```
Comment (tataihono) [inline apps/api-videos/db/seed.ts:87]: required

File: apps/api-videos/db/seed.ts [hunk 3]
```diff
@@ -67,21 +68,23 @@ interface Download {
...
 
 interface Video {
+  primaryLanguageId: string
   title: Translation[]
   snippet: Translation[]
   description: Translation[]
   studyQuestions: Translation[][]
   image: string
-  variants: VideoVariant[]
+  variants?: VideoVariant[]
```
Comment (tataihono) [inline apps/api-videos/db/seed.ts:85]: required

File: apps/api-videos/db/seed.ts [hunk 4]
```diff
@@ -67,21 +68,23 @@ interface Download {
...
 interface VideoVariant {
   id: string
-  subtitle: Translation[]
-  hls: string
+  subtitle?: Translation[]
+  hls?: string
   languageId: string
   duration: number
-  downloads: Download[]
+  downloads?: Download[]
```
Comment (tataihono) [inline apps/api-videos/db/seed.ts:75]: required

File: apps/api-videos/db/seed.ts [hunk 5]
```diff
@@ -67,21 +68,23 @@ interface Download {
 
 interface VideoVariant {
   id: string
-  subtitle: Translation[]
-  hls: string
+  subtitle?: Translation[]
```
Comment (tataihono) [inline apps/api-videos/db/seed.ts:71]: required

File: apps/api-videos/db/seed.ts [hunk 6]
```diff
@@ -265,43 +348,57 @@ async function main(): Promise<void> {
...
+            }
+          },
+          title: {
+            fields: {
+              value: {
+                analyzers: ['text_en']
               }
             }
+          },
+          isInnerSeries: {
```
Comment (tataihono) [inline apps/api-videos/db/seed.ts:384]: use `playlistIds:`

File: apps/api-videos/src/app/modules/video/video.graphql [hunk 7]
```diff
@@ -11,11 +11,19 @@ type Video @key(fields: "id primaryLanguageId") {
   studyQuestions(languageId: ID, primary: Boolean): [Translation!]!
   image: String
   variantLanguages: [Language!]!
+  """
+  Videos containing episodes are considered playlists
```
Comment (tataihono) [inline apps/api-videos/src/app/modules/video/video.graphql:15]:
```suggestion
  A Video containing episodes is considered a playlist
```

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/api-videos/src/app/modules/video/video.graphql [hunk 8]
```diff
@@ -2,20 +2,33 @@ extend type Language @key(fields: "id") {
...
   title(languageId: ID, primary: Boolean): [Translation!]!
   snippet(languageId: ID, primary: Boolean): [Translation!]!
   description(languageId: ID, primary: Boolean): [Translation!]!
   studyQuestions(languageId: ID, primary: Boolean): [Translation!]!
   image: String
   variantLanguages: [Language!]!
+  """
+  Episodes are child videos, currently only found in a playlist type
+  """
+  episodeIds: [ID!]
```
Comment (tataihono) [inline apps/api-videos/src/app/modules/video/video.graphql:24]:
```suggestion
  episodeIds: [ID!]!
```

File: apps/api-videos/src/app/modules/video/video.graphql [hunk 9]
```diff
@@ -2,20 +2,33 @@ extend type Language @key(fields: "id") {
...
+  episodes: [Video!]
```
Comment (tataihono) [inline apps/api-videos/src/app/modules/video/video.graphql:25]:
```suggestion
  episodes: [Video!]!
```

**Review by tataihono — CHANGES_REQUESTED** [review]
episodes and episodeIds should just return an empty array if there is nothing there. Otherwise every consumer is going to have to do null checks

**Review by tataihono — APPROVED** [review]

**Pull request merged**